test/unit/test-atomic.cpp

```diff
@@ -100,91 +100,524 @@ void testAtomicFunctionPol()
   testAtomicFunctionBasic<ExecPolicy, AtomicPolicy, double, 10000>();
 }
 
+template < typename T, typename AtomicPolicy >
+struct PreIncCountOp {
+  PreIncCountOp(T* count, RAJA::RangeSegment seg)
+    : counter(count), min((T)0), max((T)seg.size()-(T)1), final((T)seg.size())
+  { count[0] = (T)0; }
+  RAJA_HOST_DEVICE
+  T operator()(RAJA::Index_type RAJA_UNUSED_ARG(i)) const {
+    return (++counter) - (T)1;
+  }
+  RAJA::atomic::AtomicRef<T, AtomicPolicy> counter;
+  T min, max, final;
+};
```

Comment:
@MrBurmark minor nit. Please add a blank line between each of the struct definitions in this file so it's easier to read.